marshal/consumer.go

```diff
@@ -77,49 +76,61 @@ type Consumer struct {
 	// claims maps partition IDs to claim structures. The lock protects read/write
 	// access to this map.
 	lock   sync.RWMutex
-	claims map[int]*claim
+	claims map[string]map[int]*claim
 }
 
 // NewConsumer instantiates a consumer object for a given topic. You must create a
 // separate consumer for every individual topic that you want to consume from. Please
 // see the documentation on ConsumerBehavior.
-func (m *Marshaler) NewConsumer(topicName string, options ConsumerOptions) (*Consumer, error) {
-	// Do not allow construction post-termination
+func (m *Marshaler) NewConsumer(topicNames []string, options ConsumerOptions) (*Consumer, error) {
 	if m.Terminated() {
 		return nil, errors.New("Marshaler has terminated, no new consumers can be created")
 	}
 
-	// Ensure the user hasn't set a claim limit and put us into topic mode
-	if options.ClaimEntireTopic && options.MaximumClaims != 0 {
-		return nil, errors.New("ClaimEntireTopic and MaximumClaims are incompatible options")
+	if len(topicNames) > 1 && !options.ClaimEntireTopic {
+		return nil, errors.New("ClaimEntireTopic must be set for more than one topic")
```

Comment:
Is this a code restriction somehow?

In theory it should be possible to claim arbitrary partitions from any number of given topics.